integration-tests/chain-signatures/tests/cases/mod.rs

```diff
@@ -99,6 +100,41 @@ async fn test_signature_basic() -> anyhow::Result<()> {
     .await
 }
 
+#[test(tokio::test)]
+async fn test_signature_multiple_requests() -> anyhow::Result<()> {
+    let mut rng = rand::rngs::StdRng::from_seed([0; 32]);
+
+    with_multichain_nodes(MultichainConfig::default(), |ctx| {
+        Box::pin(async move {
+            let state_0 = wait_for::running_mpc(&ctx, Some(0)).await?;
+            assert_eq!(state_0.participants.len(), 3);
+
+            for i in 0..10 {
+                let random_secs: u32 = rng.gen_range(1, 40);
+                tokio::time::sleep(std::time::Duration::from_secs(random_secs as u64)).await;
+                let (_account, signer) = actions::new_account(&ctx).await.unwrap();
+
+                let sig_amount = rng.gen_range(1, 3);
+                tracing::info!(i, sig_amount, "Producing signature");
+
+                let (_payload_hashes, tx_hashes): (Vec<_>, Vec<_>) =
+                    actions::request_sign_multiple(&ctx, &signer, sig_amount)
+                        .await?
+                        .into_iter()
+                        .unzip();
+
+                match wait_for::signatures_responded(&ctx, &tx_hashes).await {
```

Comment:
From what I see in `signatures_responded()` fn there is no guarantee that it will return all signatures. Some may be in progress. Should we check the length here? Or make sure we have N signatures in the function?